spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -83,6 +84,7 @@ public StructType outputType() {
     Long olderThanMillis = args.isNullAt(1) ? null : DateTimeUtil.microsToMillis(args.getLong(1));
     Integer retainLastNum = args.isNullAt(2) ? null : args.getInt(2);
     Integer maxConcurrentDeletes = args.isNullAt(3) ? null : args.getInt(3);
+    Boolean streamResult = args.isNullAt(4) ? false : args.getBoolean(4);
```

Comment:
nit: can this be a primitive `boolean`?

Reply:
Yeah - let me change it to primitive `boolean` and add the related test cases.

Reply:
@aokolnychyi Just updated the PR to include the suggested the change and test cases.
For the test cases
- the original `testUseLocalIterator` method might be a legacy test case (right now it uses `collectAsList` by default), so I just renamed the method
- Use `spark.sql.adaptive.enabled=false` conf to reduce the AQE impact since spark 3.2 enables this feature by default